internal/util/chanutil/debounced_chan.go

```diff
@@ -11,45 +11,66 @@ import (
 // subsequent calls are delayed until the cooldown period has elapsed and are
 // also coalesced into a single call.
 type DebouncedChan struct {
-	done     <-chan struct{}
-	c        chan struct{}
-	cooldown time.Duration
-
-	mu                   sync.Mutex
-	timer                *time.Timer
-	sendWhenTimerExpired bool
+	c                  chan struct{}
+	cooldown           time.Duration
+	ctxDone            <-chan struct{}
+	mu                 sync.Mutex
+	sendOnTimerExpired bool
+	timer              *time.Timer
+	timerDone          bool
```

Comment:
Do you want to preserve the traditional mutex organization where its declaration precedes all the things that it protects? I think it happens to actually be alphabetical here, but maybe we should keep the line gap and add a comment about `// protects the following` or w/e

Reply:
Wait, that's a thing??

Reply:
It's a convention in the Go source code, yeah 😆  there are even linters for it https://github.com/go-critic/go-critic/issues/465

Reply:
Wow, okay that's a total TIL for me! I changed it back, but also added a comment.